db/migrate/20160917235917_create_domain_entries_table.rb

```diff
@@ -0,0 +1,17 @@
+class CreateDomainEntriesTable < ActiveRecord::Migration[5.0]
+  def self.up
+    create_table :domain_entries do |t|
+      t.string :key
```

Comment:
The intention here is that `key` is one of the `data_types` ... which might be a better name for it :P